data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -308,6 +308,32 @@ public SafeFuture<Optional<ObjectAndMetaData<List<StateValidatorData>>>> getStat
         stateIdParam, state -> getFilteredValidatorList(state, validators, statusFilter));
   }
 
+  public SafeFuture<Optional<ObjectAndMetaData<Optional<Bytes32>>>> getRandaoAtEpoch(
+      final String stateIdParam, final Optional<UInt64> epoch) {
+    return fromState(stateIdParam, state -> getRandaoAtEpochFromState(state, epoch));
+  }
+
+  @VisibleForTesting
+  Optional<Bytes32> getRandaoAtEpochFromState(
+      final tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState state,
+      final Optional<UInt64> maybeEpoch) {
+    final UInt64 stateEpoch = spec.computeEpochAtSlot(state.getSlot());
+    final int epochsPerHistoricalVector =
+        spec.atEpoch(stateEpoch).getConfig().getEpochsPerHistoricalVector();
+    final UInt64 epoch = maybeEpoch.orElseGet(() -> spec.computeEpochAtSlot(state.getSlot()));
+    if (epoch.isGreaterThan(stateEpoch)) {
+      return Optional.empty();
+    } else if (epoch.isLessThan(stateEpoch)
+        && stateEpoch.minusMinZero(epochsPerHistoricalVector).isGreaterThan(0)
+        && stateEpoch.minusMinZero(epochsPerHistoricalVector).isGreaterThanOrEqualTo(epoch)) {
+      // ignoring the first period of epoch=0 to epochsPerHistoricalVector,
+      // return empty if the epoch is not within `epochsPerHistoricalVector` of the state epoch
+      return Optional.empty();
+    }
+    return Optional.of(
+        spec.atSlot(state.getSlot()).beaconStateAccessors().getRandaoMix(state, epoch));
```

Comment:
On https://github.com/ethereum/beacon-APIs/pull/242 part of the reason for having this method seems to be related to not loading the entire state for an epoch just to get the RANDAO. However, what we are doing here seems to be just hiding the fact that we are loading the state to get the RANDAO.

This is better than returning the whole state over the wire, but if most of the time spent in this operation is in the actual loading of the state, should we try to address this?

Reply:
we're separating business logic from interface basically, just how our rest api's are structured, we can't really do anything but load the state...

Reply:
its definitely a lot less effort over the wire to xmit 1 bytes32 payload than the entire state... i think that's the argument for the endpoint...